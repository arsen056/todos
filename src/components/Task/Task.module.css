.taskWrapper {
  display: flex;
  justify-content: space-between;
  margin: 1rem 0;
  box-shadow: 0 -1px 5px 0 rgba(0,0,0,0.3);
  padding: 0.5rem 1rem;
}

.task {
  cursor: pointer;
  display: inline;
}

.taskItem {
  display: flex;
}

.checked {
  text-decoration: line-through;
  opacity: 0.5;
}

.checkbox {
  display: none;
}

.customCheckbox {
  position: relative;
  display: inline-block;
  width: 30px;
  height: 30px;
  background: #FFFFFF;
  border: 2px solid #7E9BBD;
  border-radius: 4px;
  vertical-align: sub;
  margin-right: 5px;
}

.customCheckbox::before {
  content: '';
  display: inline-block;
  width: 23px;
  height: 23px;
  background-image: url('../../common/assets/done.svg');
  background-size: contain;
  background-repeat: no-repeat;
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%) scale(0);
  transition: 0.2s ease-in;
}

.checkbox:checked +.customCheckbox::before {
  transform: translate(-50%, -50%) scale(1);
}