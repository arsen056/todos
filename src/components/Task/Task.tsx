import React, {FC} from 'react';

type Props = {
  title: string
  isDone: boolean
}
export const Task:FC<Props> = ({title, isDone}) => {
  return (
    <li>
      <input checked={isDone} type="checkbox"/>
      <span>{title}</span>
    </li>
  );
};
