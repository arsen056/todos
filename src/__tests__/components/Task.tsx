import {render} from '@testing-library/react';
import {Task} from 'components/Task/Task';

describe('Task', () => {
  it('Should be render task isDone: true', () => {
    const view = render(<Task id={'1'} title={'Task'} isDone={true} />);
    expect(view).toMatchSnapshot();
  });

  it('Should be render task isDone: false', () => {
    const view = render(<Task id={'1'} title={'Task'} isDone={false} />);
    expect(view).toMatchSnapshot();
  });
});
