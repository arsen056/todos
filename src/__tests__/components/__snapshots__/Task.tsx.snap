// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Should be render task isDone: false 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="taskWrapper"
      >
        <div
          class="taskItem"
        >
          <label
            class="task "
          >
            <input
              class="checkbox"
              type="checkbox"
            />
            <span
              class="customCheckbox"
            />
          </label>
          <span
            class="span "
          >
            Task
          </span>
        </div>
        <button
          class="button undefined"
        >
          x
        </button>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="taskWrapper"
    >
      <div
        class="taskItem"
      >
        <label
          class="task "
        >
          <input
            class="checkbox"
            type="checkbox"
          />
          <span
            class="customCheckbox"
          />
        </label>
        <span
          class="span "
        >
          Task
        </span>
      </div>
      <button
        class="button undefined"
      >
        x
      </button>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Task Should be render task isDone: true 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="taskWrapper"
      >
        <div
          class="taskItem"
        >
          <label
            class="task checked"
          >
            <input
              checked=""
              class="checkbox"
              type="checkbox"
            />
            <span
              class="customCheckbox"
            />
          </label>
          <span
            class="span checked"
          >
            Task
          </span>
        </div>
        <button
          class="button undefined"
        >
          x
        </button>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="taskWrapper"
    >
      <div
        class="taskItem"
      >
        <label
          class="task checked"
        >
          <input
            checked=""
            class="checkbox"
            type="checkbox"
          />
          <span
            class="customCheckbox"
          />
        </label>
        <span
          class="span checked"
        >
          Task
        </span>
      </div>
      <button
        class="button undefined"
      >
        x
      </button>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Task Should be render todolist 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="taskWrapper"
      >
        <div
          class="taskItem"
        >
          <label
            class="task checked"
          >
            <input
              checked=""
              class="checkbox"
              type="checkbox"
            />
            <span
              class="customCheckbox"
            />
          </label>
          <span
            class="span checked"
          >
            Task
          </span>
        </div>
        <button
          class="button undefined"
        >
          x
        </button>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="taskWrapper"
    >
      <div
        class="taskItem"
      >
        <label
          class="task checked"
        >
          <input
            checked=""
            class="checkbox"
            type="checkbox"
          />
          <span
            class="customCheckbox"
          />
        </label>
        <span
          class="span checked"
        >
          Task
        </span>
      </div>
      <button
        class="button undefined"
      >
        x
      </button>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
